core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -303,17 +306,19 @@ public void run() {
       return jumpListener;
     }
 
-    private Future<?> scheduleTimeout(final ServerStream stream, Metadata headers) {
+    private Future<?> scheduleTimeout(final ServerStream stream, Metadata headers,
+                                      final Context.CancellableContext context) {
       Long timeoutMicros = headers.get(TIMEOUT_KEY);
       if (timeoutMicros == null) {
         return DEFAULT_TIMEOUT_FUTURE;
       }
-      return timeoutService.schedule(new Runnable() {
+      return timeoutService.schedule(new ContextRunnable(context) {
```

Comment:
Calling the transport doesn't seem to be something where the context needs to set.